### ISSUE TITLE
Data Source: TestGrid Metadata

### ISSUE THREAD

**Issue by MichaelClifford** [aicoe-aiops/ocp-ci-analysis#144]
We need to update the `testgrid_EDA.ipynb' notebook and data collection steps to include relevant meta data from the testgrids that is currently not being captured. 

We initially only collected the "statuses" values from the "tests" field in order to recreate the 2D test grids themselves, and made significant initial progress working directly with that data.      After reviewing some of the KPI metrics definitions it is apparent that additional data is likely required. Before, going to any external sources, let's take another look at the additional test grid data and collect it for future use.  

**Acceptance Criteria** 

- [ ] Extend `testgrid_EDA.ipynb' (or make a new notebook if cleaner) going through the additional data fields provided by testgrid.  
- [ ] Create method for collecting extra data
- [ ] Save sample data set

**Comment by hemajv** [comment]
While exploring the testgrid metadata, I looked more closely into the `overall-status` field for the `openshift-ocp-release-4.2-informing` dashboard for the job `release-openshift-ocp-e2e-aws-scaleup-rhel7-4.2`: https://testgrid.k8s.io/redhat-openshift-ocp-release-4.2-informing#release-openshift-origin-installer-e2e-aws-upgrade-rollback-4.1-to-4.2. 

The `overall-status` field returns a status code for the job, it returned `3` for this particular job.

![Screenshot from 2021-03-09 12-11-06](https://user-images.githubusercontent.com/7343099/110509887-9c5fa980-80d0-11eb-960f-d9a7ffee0956.png)

Seems like these status codes map to those defined [here](https://github.com/GoogleCloudPlatform/testgrid/blob/1dbc0e8b6ef997b9c90b3fb79518c35ce4d27ea2/pb/summary/summary.proto#L163), and this job would be a "Failing" one. This differs from the status codes defined [here](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/docs/cell_labels_in_testgrid.md), which seem to be more related to the individual tests themselves.

If we run it for the job `promote-release-openshift-machine-os-content-e2e-aws-4.2-s390x`: https://testgrid.k8s.io/redhat-openshift-ocp-release-4.2-informing#promote-release-openshift-machine-os-content-e2e-aws-4.2-s390x, the `overall-status` code is `2` which is "Pass", and I think this is interesting in spite of few failing tests observed in the job:

![Screenshot from 2021-03-09 12-26-37](https://user-images.githubusercontent.com/7343099/110511894-ba2e0e00-80d2-11eb-9778-ffaa7fde2352.png)

**Comment by hemajv** [comment]
Ah so the `overall-status` being returned also corresponds to how it is displayed here: https://testgrid.k8s.io/redhat-openshift-ocp-release-4.2-informing

if we look at the dashboard `redhat-openshift-ocp-release-4.2-informing`, it lists out all the jobs like so (there is also a `Sort by status` filter) and since the `overall-status` for `promote-release-openshift-machine-os-content-e2e-aws-4.2-s390x` is "Pass", the `overall-status` field returns the value `2`:

![Screenshot from 2021-03-09 14-53-58](https://user-images.githubusercontent.com/7343099/110529652-84475480-80e7-11eb-8a66-d4b555cf3ec5.png)

**Comment by hemajv** [comment]
I don't think it addresses the problem. Still trying to figure out how the "build pass/fail" fields are being defined/used. As per my understanding, the `overall_status` field just informs the overall status for the entire "job" itself. So in the example of https://testgrid.k8s.io/redhat-openshift-ocp-release-4.2-informing#promote-release-openshift-machine-os-content-e2e-aws-4.2-s390x, the `overall_status` is `2` indicating that job `promote-release-openshift-machine-os-content-e2e-aws-4.2-s390x` is considered as a "Passing" job. I still think we need to explore the metadata more to figure out where can we accurately extract "build pass/fail" values.

**Comment by MichaelClifford** [comment]
I see now @hemajv, your right, I don think it solves our immediate issue as its making some aggregate claim about the overall job overtime. Thanks for the clarification.

**Comment by oindrillac** [comment]
@hemajv and I collaborating on my fork and added a separate notebook: https://github.com/oindrillac/ocp-ci-analysis/blob/metadata-eda/notebooks/data-sources/TestGrid/testgrid_metadata_EDA.ipynb to tackle this issue. Please feel free to contribute there if you also plan on working on this. :slightly_smiling_face: 

cc: @aakankshaduggal @chauhankaranraj

**Comment by chauhankaranraj** [comment]
> @hemajv and I collaborating on my fork and added a separate notebook: https://github.com/oindrillac/ocp-ci-analysis/blob/metadata-eda/notebooks/data-sources/TestGrid/testgrid_metadata_EDA.ipynb to tackle this issue. Please feel free to contribute there if you also plan on working on this. 

Awesome, in that case @aakankshaduggal and I can do the EDA for the rest of fields and push our updates here :)

**Comment by oindrillac** [comment]
Thanks @chauhankaranraj! My fork is up to date with our changes, please feel free to add the rest of the fields

**Comment by MichaelClifford** [comment]
@oindrillac do you want to just push your existing notebook into a "stage" directory under `notebooks/data-sources/TestGrid` that way everyone can work off the main repo instead of forking forks.  WDYT?

@chauhankaranraj @hemajv

**Comment by oindrillac** [comment]
@MichaelClifford sounds good :+1:

**Comment by aakankshaduggal** [comment]
While exploring testgrid metadata, @chauhankaranraj and I came across the field `triage-enabled`. On further inspection of the same, we discovered that in addition to the [testgrid](https://testgrid.k8s.io/) dashboard, Google has another dashboard called [triage](https://go.k8s.io/triage). This dashboard "shows clusters of similar test failures across all jobs", and suggests what general subject these failures related to e.g. storage. So this the value, **True** or **False**, returned in this field probably tells whether triage is enabled or not for this particular dashboard/tab.

It could be interesting to look into the work performed by [Triage](https://github.com/kubernetes/test-infra/tree/master/triage) for our future work.

cc : @MichaelClifford
